### ISSUE TITLE
[Feature] Support Skia Shaders for Cool Effects?

### ISSUE THREAD

**Issue by NateAGeek** [neovide/neovide#1670]
**Is your feature request related to a problem? Please describe.**
I was watching a nice video on YouTube:
[![Video](https://img.youtube.com/vi/lr7YpK6rrAM/0.jpg)](https://www.youtube.com/watch?v=lr7YpK6rrAM)
They had a cool glow effect, that looks to be a post-process effect on the video... Making this effect normally just not possible... However, I saw that this project uses Skia and could whip up a sample shader to see if I can get this to work. I did, but it is hacked and I am compiling the shader on every frame, lol. 

**Describe the solution you'd like**
Maybe allow the user to store some sksl files in their user directory, like a .neovide_shaders folder or something... Then expose the setting as a config like you have?

**Additional context**
Happy to take on this as a small project on the side, as I would like to get my feet wet more in graphics/rust. But just thought I'd see if you have any opinions before I really dig into making a full pr.

Here is my output, and I'll make a PR example soon.
<img width="2198" alt="image" src="https://user-images.githubusercontent.com/1813055/206971022-7addd185-2f1f-4719-a807-84ca968144dc.png">

**Comment by NateAGeek** [comment]
#1671
That is the PR with the sample code, feel free to tell me any feedback.

**Comment by shaunsingh** [comment]
Demo looks cool! It would be a nice addition to have some shaders for neovim, give it a cool-retro-term kindof look 

I think it would be a good idea to precompile shaders rather than recompile every frame, if thats possible

**Comment by NateAGeek** [comment]
Yeah, talked a bit with @MultisampledNight on some ideas on config. I might look into adding in a setting in the RendererSettings... But I need to take some time and see. Plus, not sure if to build some in source or what not... Just when I have time I might get to it.

**Comment by p4-k4** [comment]
Awakening this!

**Comment by fredizzimo** [comment]
This will be much easier to support once we migrate to https://github.com/neovide/vide

But, it's still far from being ready and stable.
